webapp/components/PostCard/PostCard.vue

```diff
@@ -143,23 +148,26 @@ export default {
   },
 }
 </script>
-
-<style scoped lang="scss">
-.ds-card-image img {
-  width: 100%;
-  max-height: 2000px;
-  object-fit: contain;
```

Comment:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-13T12:09:36Z" title="Friday, December 13th 2019, 1:09:36 pm +01:00">Dec 13, 2019</time>_

---
> **Outdated (history rewrite)** - original diff
---
```diff
@@ -141,25 +146,37 @@ export default {
       this.$emit('unpinPost', post)
     },
   },
-}
-</script>
+  mounted() {
+    const width = this.$el.offsetWidth
+    const height = Math.min(width / this.post.imageAspectRatio, 2000)
+    const imageElement = this.$el.querySelector('.ds-card-imag')
```


```suggestion
    const imageElement = this.$el.querySelector('.ds-card-image')
```

Reply:
> [<img alt="mattwr18" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/mattwr18) **Authored by [mattwr18](https://github.com/mattwr18)**
_<time datetime="2019-12-13T12:10:50Z" title="Friday, December 13th 2019, 1:10:50 pm +01:00">Dec 13, 2019</time>_

---


Also, I'm concerned about this because we have not introduced `imageAspectRatio` for the overwhelming majority of `Post`... new `Posts` add their `imageAspectRatio`, but we would need to convert more than 18,000 `Post` where this is null... so what happens in that case?

Reply:
> [<img alt="ogerly" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-12-13T12:42:14Z" title="Friday, December 13th 2019, 1:42:14 pm +01:00">Dec 13, 2019</time>_

---


@mattwr18  i have change .ds-card-imag(e)

Reply:
> [<img alt="alina-beck" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/alina-beck) **Authored by [alina-beck](https://github.com/alina-beck)**
_<time datetime="2019-12-16T10:07:24Z" title="Monday, December 16th 2019, 11:07:24 am +01:00">Dec 16, 2019</time>_

---


I don't know how this code made its way into this PR – maybe from merging in `master` while we had it in there? – but it's breaking the layout. This is what happens to images without aspect ratio:
<img width="226" alt="Screenshot 2019-12-16 at 15 30 41" src="https://user-images.githubusercontent.com/11694483/70897737-3abda600-2019-11ea-89cd-debdc0d5bd66.png">

Also, weirdly, when I edit a `Post` – even without changing the picture or `blurred` status the post loses its `imageAspectRatio`. So unfortunately there's more cleaning up to do! 😞

Reply:
> [<img alt="ogerly" height="40" width="40" align="left" src="https://avatars1.githubusercontent.com/u/38261864?s=88&v=4">](https://github.com/ogerly) **Authored by [ogerly](https://github.com/ogerly)**
_<time datetime="2019-12-17T06:37:49Z" title="Tuesday, December 17th 2019, 7:37:49 am +01:00">Dec 17, 2019</time>_

---


![cheetah-rain](https://user-images.githubusercontent.com/1324583/70970963-13f68280-20a0-11ea-88d0-b827b28972b3.gif)